azure-iot-device/azure/iot/device/iothub/pipeline/mqtt_topic_iothub.py

```diff
@@ -252,3 +252,38 @@ def encode_properties(message_to_send, topic):
         topic += user_properties_encoded
 
     return topic
+
+
+def get_twin_response_topic_for_subscribe():
+    return "$iothub/twin/res/#"
+
+
+def get_twin_patch_topic_for_subscribe():
+    "$iothub/twin/PATCH/properties/desired/#"
+
+
+def get_twin_topic_for_publish(verb, resource, request_id):
+    return "$iothub/twin/{}{}?$rid={}".format(verb, resource, request_id)
+
+
+def is_twin_response_topic(topic):
+    return topic.startswith("$iothub/twin/res/")
+
+
+def get_twin_request_id_from_topic(topic):
+    parts = topic.split("/")
+    if is_twin_response_topic(topic) and len(parts) >= 4:
+        properties = _extract_properties(topic.split("?")[1])
```

Comment:
>topic.split("?")[1] [](start = 41, length = 19)

I wonder if we should move the `topic.split("?")[1]` logic inside `_extract_properties`, since it's always kind of needed.